src/types/DashboardVisualisation.ts

```diff
@@ -26,7 +26,9 @@ export enum DashboardItemSize {
 }
 
 export type DashboardVisualisation = {
+    id?: string;
```

Comment:
Is this optional?

Reply:
Yes :D

Reply:
The Visualisation on the dashboard needs to have an `id`. Is it possible to rewrite the code such that this type can have a `NonNullable` `id`?

Reply:
Perhaps by immediately assigning an `id` field in the cases where it's not present. I.e. move it from the `AddToDashboard` atom to somewhere else

Reply:
I think it would be cleaner if the Dashboard didn't have to care about _how_ the objects are before they end up on the dashboard.

Reply:
Trying to fix this =)